pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -813,13 +817,17 @@ func (r *githubRepository) updateWebhook(ctx context.Context) (pgh.WebhookConfig
 		return hook, false, nil
 	}
 
+	// If we need a new webhook, always update the secret as well.
```

Comment:
this is an update not a create, isn't it?

Reply:
Yep. That code path handles when the webhook is deleted